packages/gatsby/src/redux/run-sift.js

```diff
@@ -64,7 +64,8 @@ function handleFirst(siftArgs, nodes) {
       )
 
   if (index !== -1) {
-    return [nodes[index]]
+    const result  = nodes[index];
+    return result ? [result] : [];
```

Comment:
Instead of doing this, I'd suggest adding a special-case check at the top of this function:

```js
if (nodes.length === 0) {
  return []
}
```

As the old code in general looks correct, except that it assumes `nodes` is not empty.

Reply:
OK - I will do it this way.